File: lowcode-commons-parent/lowcode-commons-descriptor/src/main/java/io/github/cloudstars/lowcode/commons/descriptor/Descriptor.java
package io.github.cloudstars.lowcode.commons.descriptor;

import java.util.List;

/**
 * 规范对象（先有规范，通过规范实例化为配置）
 *
 * @author clouds
 */
public class Descriptor implements XDescriptor {

    /**
     * 规范的名称
     */
    private String name;

    /**
     * 规范的类名
     *
     * @return
     */
    private String configClassName;

    /**
     * 规范的属性
     */
    private List<ConfigAttribute> attributes;

    @Override
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String getConfigClassName() {
        return configClassName;
    }

    public void setConfigClassName(String configClassName) {
        this.configClassName = configClassName;
    }

    @Override
    public List<ConfigAttribute> getAttributes() {
        return attributes;
    }

    public void setAttributes(List<ConfigAttribute> attributes) {
        this.attributes = attributes;
    }

}

File: lowcode-commons-parent/lowcode-commons-descriptor/src/test/resources/descriptors/descriptor.js
/**
 * 规范定义函数
 *
 * @param options
 * @returns {{name: string, description: string, attributes: [{dataType: string, name: string},{dataType: string, name: string, properties: [{dataType: string, name: string},{dataType: string, name: string}]},{dataType: string, name: string, items: {__ref: string, dataType: string}}], configClassName: string}}
 */
(options) => {
    return {
        "name": "descriptor",
        "description": "这是一个规范，对应的配置类型是SomeConfig",
        "configClassName": "io.github.cloudstars.lowcode.commons.descriptor.SomeConfig",
        "attributes": [
            {
                "name": "x",
                "dataType": "STRING",
            },
            {
                "name": "properties",
                "dataType": "OBJECT",
                "properties": [{
                    "name": "aaa",
                    "dataType": "STRING",
                }, {
                    "name": "bbb",
                    "dataType": "STRING",
                }]
            },
            {
                "name": "items",
                "dataType": "ARRAY",
                "items": {
                    "dataType": "REFERENCE",
                    "descriptor": "descriptorRef",
                }
            }
        ]
    };
};